File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/orderstatus/TradeItFillParcelable.java
package it.trade.android.sdk.model.orderstatus;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import it.trade.model.reponse.Fill;

public class TradeItFillParcelable implements Parcelable {

    @SerializedName("timestampFormat")
    private String timestampFormat;

    @SerializedName("price")
    private Double price;

    @SerializedName("timestamp")
    private String timestamp;

    @SerializedName("quantity")
    private Integer quantity;

    TradeItFillParcelable(Fill fill) {
        this.timestampFormat = fill.timestampFormat;
        this.price = fill.price;
        this.timestamp = fill.timestamp;
        this.quantity = fill.quantity;
    }

    public String getTimestampFormat() {
        return timestampFormat;
    }

    public Double getPrice() {
        return price;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public Integer getQuantity() {
        return quantity;
    }

    @Override
    public String toString() {
        return "TradeItFillParcelable{" +
                "timestampFormat='" + timestampFormat + '\'' +
                ", price=" + price +
                ", timestamp='" + timestamp + '\'' +
                ", quantity=" + quantity +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TradeItFillParcelable that = (TradeItFillParcelable) o;

        if (timestampFormat != null ? !timestampFormat.equals(that.timestampFormat) : that.timestampFormat != null)
            return false;
        if (price != null ? !price.equals(that.price) : that.price != null) return false;
        if (timestamp != null ? !timestamp.equals(that.timestamp) : that.timestamp != null)
            return false;
        return quantity != null ? quantity.equals(that.quantity) : that.quantity == null;

    }

    @Override
    public int hashCode() {
        int result = timestampFormat != null ? timestampFormat.hashCode() : 0;
        result = 31 * result + (price != null ? price.hashCode() : 0);
        result = 31 * result + (timestamp != null ? timestamp.hashCode() : 0);
        result = 31 * result + (quantity != null ? quantity.hashCode() : 0);
        return result;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.timestampFormat);
        dest.writeValue(this.price);
        dest.writeString(this.timestamp);
        dest.writeValue(this.quantity);
    }

    protected TradeItFillParcelable(Parcel in) {
        this.timestampFormat = in.readString();
        this.price = (Double) in.readValue(Double.class.getClassLoader());
        this.timestamp = in.readString();
        this.quantity = (Integer) in.readValue(Integer.class.getClassLoader());
    }

    public static final Parcelable.Creator<TradeItFillParcelable> CREATOR = new Parcelable.Creator<TradeItFillParcelable>() {
        @Override
        public TradeItFillParcelable createFromParcel(Parcel source) {
            return new TradeItFillParcelable(source);
        }

        @Override
        public TradeItFillParcelable[] newArray(int size) {
            return new TradeItFillParcelable[size];
        }
    };
}

File: tradeit-android-sdk/src/main/kotlin/it/trade/android/sdk/model/orderstatus/TradeItFillParcelable.kt
package it.trade.android.sdk.model.orderstatus

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.SerializedName

import it.trade.model.reponse.Fill

class TradeItFillParcelable : Parcelable {

    @SerializedName("timestampFormat")
    var timestampFormat: String? = null
        private set

    @SerializedName("price")
    var price: Double? = null
        private set

    @SerializedName("timestamp")
    var timestamp: String? = null
        private set

    @SerializedName("quantity")
    var quantity: Int? = null
        private set

    internal constructor(fill: Fill) {
        this.timestampFormat = fill.timestampFormat
        this.price = fill.price
        this.timestamp = fill.timestamp
        this.quantity = fill.quantity
    }

    override fun toString(): String {
        return "TradeItFillParcelable{" +
                "timestampFormat='" + timestampFormat + '\''.toString() +
                ", price=" + price +
                ", timestamp='" + timestamp + '\''.toString() +
                ", quantity=" + quantity +
                '}'.toString()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false

        val that = other as TradeItFillParcelable?

        if (if (timestampFormat != null) timestampFormat != that!!.timestampFormat else that!!.timestampFormat != null)
            return false
        if (if (price != null) price != that.price else that.price != null) return false
        if (if (timestamp != null) timestamp != that.timestamp else that.timestamp != null)
            return false
        return if (quantity != null) quantity == that.quantity else that.quantity == null

    }

    override fun hashCode(): Int {
        var result = if (timestampFormat != null) timestampFormat!!.hashCode() else 0
        result = 31 * result + if (price != null) price!!.hashCode() else 0
        result = 31 * result + if (timestamp != null) timestamp!!.hashCode() else 0
        result = 31 * result + if (quantity != null) quantity!!.hashCode() else 0
        return result
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.timestampFormat)
        dest.writeValue(this.price)
        dest.writeString(this.timestamp)
        dest.writeValue(this.quantity)
    }

    protected constructor(`in`: Parcel) {
        this.timestampFormat = `in`.readString()
        this.price = `in`.readValue(Double::class.java.getClassLoader()) as? Double
        this.timestamp = `in`.readString()
        this.quantity = `in`.readValue(Int::class.java.getClassLoader()) as? Int
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<TradeItFillParcelable> = object : Parcelable.Creator<TradeItFillParcelable> {
            override fun createFromParcel(source: Parcel): TradeItFillParcelable {
                return TradeItFillParcelable(source)
            }

            override fun newArray(size: Int): Array<TradeItFillParcelable?> {
                return arrayOfNulls(size)
            }
        }
    }
}
